Mode: Scraping - cant handle non numeric values (e.g. On / Off)

When attempting to start the Docker container with the ISG_URL, ISG_PASSWORD, and ISG_USER variables, the container stops after a few seconds with the following log message:

panic: runtime error: index out of range [1] with length 0
goroutine 22 [running]:
main.normalizeValue({0xc000704c99, 0x3})
/build/scraping.go:150 +0x2c6
main.parsePage.func1(0xc0000120f8?, 0x0?)
/build/scraping.go:110 +0xfb
github.com/PuerkitoBio/goquery.(*Selection).Each(0xc0003af890, 0xc00060fd08)
/go/pkg/mod/github.com/!puerkito!bio/goquery@v1.8.1/iteration.go:10 +0x46
main.parsePage({0xba6637?, 0x0?}, 0xc00060fd70)
/build/scraping.go:95 +0x11b
main.gatherScrapingData()
/build/scraping.go:73 +0x18e
main.gatherData()
/build/main.go:167 +0x108
main.main.func1()
/build/main.go:88 +0x19
created by main.main
/build/main.go:86 +0x10b

The ISG is set to English, and the ISG version is v12.2.1.

@sfudeus 

First run prompts directly an error and let crash the container.

```
panic: runtime error: index out of range [1] with length 0

goroutine 13 [running]:
main.normalizeValue({0xc0002fa759, 0x3})
        /build/scraping.go:151 +0x25d
main.parsePage.func1(0x0?, 0xc0005cd710)
        /build/scraping.go:110 +0xe5
github.com/PuerkitoBio/goquery.(*Selection).Each(0xc0005cca80, 0xc0004d9d00)
        /go/pkg/mod/github.com/!puerkito!bio/goquery@v1.9.1/iteration.go:10 +0x62
main.parsePage({0xc44709?, 0x0?}, 0xc0004d9ec8)
        /build/scraping.go:95 +0x114
main.gatherScrapingData()
        /build/scraping.go:73 +0x19b
main.gatherData()
        /build/main.go:178 +0x10a
main.main.func1()
        /build/main.go:96 +0x13
created by main.main in goroutine 1
        /build/main.go:94 +0x111
panic: runtime error: index out of range [1] with length 0
```

I just switched to your new tag 1.4.2 and runs my docker-compose.yml. It was just a quick try out. I try to check later more details what's failing.

That was a typical brain AFK mistake.

Now I am on the right branch. I can tell you for German is working. But something else is not working when language is English.

```
panic: descriptor Desc{fqName: "isg_set_superheating,_suction_gas,_compresso", help: "Metric set_superheating,_suction_gas,_compresso in K", constLabels: {}, variableLabels: {unit}} is invalid: "isg_set_superheating,_suction_gas,_compresso" is not a valid metric name

goroutine 51 [running]:
github.com/prometheus/client_golang/prometheus.(*Registry).MustRegister(0x1292c20, {0xc0002882c0?, 0x0?, 0x0?})
        /go/pkg/mod/github.com/prometheus/client_golang@v1.19.0/prometheus/registry.go:405 +0x66
github.com/prometheus/client_golang/prometheus/promauto.Factory.NewGaugeVec({{0xd79898?, 0x1292c20?}}, {{0xc446ee, 0x3}, {0x0, 0x0}, {0xc0005311d0, 0x28}, {0xc000437e00, 0x34}, ...}, ...)
        /go/pkg/mod/github.com/prometheus/client_golang@v1.19.0/prometheus/promauto/auto.go:308 +0x163
github.com/prometheus/client_golang/prometheus/promauto.NewGaugeVec(...)
        /go/pkg/mod/github.com/prometheus/client_golang@v1.19.0/prometheus/promauto/auto.go:198
main.createOrRetrieve({0xc0005311d0, 0x28}, {0xc000402407, 0x1}, 0x0)
        /build/main.go:208 +0x610
main.parsePage.func1(0x0?, 0xc000566330)
        /build/scraping.go:118 +0x24b
github.com/PuerkitoBio/goquery.(*Selection).Each(0xc000556390, 0xc0000add00)
        /go/pkg/mod/github.com/!puerkito!bio/goquery@v1.9.1/iteration.go:10 +0x62
main.parsePage({0xc44718?, 0x0?}, 0xc0000adec8)
        /build/scraping.go:96 +0x114
main.gatherScrapingData()
        /build/scraping.go:79 +0x218
main.gatherData()
        /build/main.go:178 +0x10a
main.main.func1()
        /build/main.go:96 +0x13
created by main.main in goroutine 1
        /build/main.go:94 +0x111
```

--- a/scraping.go
+++ b/scraping.go
@@ -1,6 +1,7 @@
 package main
 
 import (
+	"fmt"
 	"regexp"
 	"strconv"
 	"strings"
@@ -107,7 +108,11 @@ func parsePage(page string, flagRemovalList map[string]prometheus.Gauge) {
 		}
 
 		if value != "" {
-			isgValue := normalizeValue(value)
+			isgValue, err := normalizeValue(value)
+			if err != nil {
+				log.Warnf("Failed to process value %s for label %s, skipping", value, label)
+				return
+			}
 			valuesMap[label] = make([]IsgValue, 0)
 			valuesMap[label] = append(valuesMap[label], isgValue)
 			createOrRetrieve(label, isgValue.Unit, nil).Set(isgValue.Value)
@@ -143,16 +148,29 @@ func normalizeLabel(s string) string {
 	return s
 }
 
-func normalizeValue(s string) IsgValue {
+func normalizeValue(s string) (IsgValue, error) {
+
+	// some values use fixed boolean vocabulary
+	switch s {
+	case "Aus", "Off", "Apagado", "Uit", "Spento", "Av", "Wyłączony", "Vyp", "Kikapcsolva", "Apagat", "Pois":
+		s = "0"
+	case "Ein", "On", "Allumé", "Aan", "Acceso", "På", "Włączony", "Zap", "Bekapcsolva", "Encendido", "Päällä", "Tændt":
+		s = "1"
+	}
+
 	re := regexp.MustCompile(`(?P<value>[0-9,.-]+)( ?)(?P<unit>[a-zA-Z°%/²³.]*)`)
 	matches := re.FindStringSubmatch(s)
+	if len(matches) == 0 {
+		return IsgValue{}, fmt.Errorf("failed to parse value %s", s)
+	}
+
 	// ISG exports numbers with decimal separator ",", even with language setting english
 	// needs to be converted to be parsed as float
 	value := strings.Replace(matches[re.SubexpIndex("value")], ",", ".", -1)
 	unit := ""
 	float, err := strconv.ParseFloat(value, 64)
 	if err != nil {
-		log.Fatalf("Failed to parse value %s", value)
+		return IsgValue{}, fmt.Errorf("failed to parse value %s: %s", value, err)
 	}
 
 	if len(matches) > 2 {
@@ -163,5 +181,5 @@ func normalizeValue(s string) IsgValue {
 		}
 	}
 
-	return IsgValue{Value: float, Unit: unit}
+	return IsgValue{Value: float, Unit: unit}, nil
 }

--- a/scraping_test.go
+++ b/scraping_test.go
@@ -143,6 +143,12 @@ func TestPage(t *testing.T) {
 				t.Errorf("Failed delivering sample for 4,7")
 			}
 			fmt.Fprint(w, string(content))
+		} else if page == "onoff" {
+			content, err := os.ReadFile("test_resources/sample_onoff.html")
+			if err != nil {
+				t.Errorf("Failed delivering sample for onoff")
+			}
+			fmt.Fprint(w, string(content))
 		} else {
 			fmt.Fprint(w, "")
 		}
@@ -158,6 +164,11 @@ func TestPage(t *testing.T) {
 	flagRemovalList := make(map[string]prometheus.Gauge)
 	parsePage("1,1", flagRemovalList)
 	parsePage("4,7", flagRemovalList)
+	parsePage("onoff", flagRemovalList)
+
+	if valuesMap["festwertbetrieb"] == nil {
+		t.Errorf("Failed to find expected onoff value")
+	}
 
 	json, _ := json.Marshal(valuesMap)
 	fmt.Println(string(json))

--- a/scraping_test.go
+++ b/scraping_test.go
@@ -12,9 +12,19 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 )
 
+func checkExpectError(err error, expect bool, t *testing.T) {
+	if err != nil && !expect {
+		t.Errorf("Expected success, got error: %s", err)
+	}
+	if err == nil && expect {
+		t.Errorf("Expected error, got none")
+	}
+}
+
 func TestNormalizeValue(t *testing.T) {
 
-	res := normalizeValue("10,123 kWh")
+	res, err := normalizeValue("10,123 kWh")
+	checkExpectError(err, false, t)
 	if res.Unit != "kWh" {
 		t.Errorf("Expected kWh as unit, but got %s", res.Unit)
 	}
@@ -22,7 +32,8 @@ func TestNormalizeValue(t *testing.T) {
 		t.Errorf("Expected 10.123 as value, but got %f", res.Value)
 	}
 
-	res = normalizeValue("10")
+	res, err = normalizeValue("10")
+	checkExpectError(err, false, t)
 	if res.Unit != "" {
 		t.Error("Expected empty unit")
 	}
@@ -30,7 +41,8 @@ func TestNormalizeValue(t *testing.T) {
 		t.Fail()
 	}
 
-	res = normalizeValue("3,345 MWh")
+	res, err = normalizeValue("3,345 MWh")
+	checkExpectError(err, false, t)
 	if res.Unit != "kWh" {
 		t.Errorf("Expected conversion to kWh, but got %s", res.Unit)
 	}
@@ -38,7 +50,8 @@ func TestNormalizeValue(t *testing.T) {
 		t.Errorf("Expected 3345, but got %f", res.Value)
 	}
 
-	res = normalizeValue("18.321kWh")
+	res, err = normalizeValue("18.321kWh")
+	checkExpectError(err, false, t)
 	if res.Unit != "kWh" {
 		t.Errorf("Expected kWh as unit, but got %s", res.Unit)
 	}
@@ -46,7 +59,8 @@ func TestNormalizeValue(t *testing.T) {
 		t.Errorf("Expected 18.321, but got %f", res.Value)
 	}
 
-	res = normalizeValue("15,8°C")
+	res, err = normalizeValue("15,8°C")
+	checkExpectError(err, false, t)
 	if res.Unit != "°C" {
 		t.Errorf("Expected °C as unit, but got %s", res.Unit)
 	}
@@ -54,17 +68,34 @@ func TestNormalizeValue(t *testing.T) {
 		t.Errorf("Expected 15.8, but got %f", res.Value)
 	}
 
-	res = normalizeValue("-15,8°C")
+	res, err = normalizeValue("-15,8°C")
+	checkExpectError(err, false, t)
 	if res.Unit != "°C" {
 		t.Errorf("Expected °C as unit, but got %s", res.Unit)
 	}
 	if res.Value != -15.8 {
 		t.Errorf("Expected -15.8, but got %f", res.Value)
 	}
-	res = normalizeValue("1 l/min")
-	res = normalizeValue("1 %")
-	res = normalizeValue("1 m³/h")
+	res, err = normalizeValue("1 l/min")
+	checkExpectError(err, false, t)
+	res, err = normalizeValue("1 %")
+	checkExpectError(err, false, t)
+	res, err = normalizeValue("1 m³/h")
+	checkExpectError(err, false, t)
 
+	res, err = normalizeValue("On")
+	checkExpectError(err, false, t)
+	if res.Value != 1 {
+		t.Errorf("Expected 1, but got %f", res.Value)
+	}
+	res, err = normalizeValue("Off")
+	checkExpectError(err, false, t)
+	if res.Value != 0 {
+		t.Errorf("Expected 0, but got %f", res.Value)
+	}
+
+	res, err = normalizeValue("fnvdnvsdbvdfk")
+	checkExpectError(err, true, t)
 }
 
 func TestNormalizeLabel(t *testing.T) {

--- a/test_resources/sample_onoff.html
+++ b/test_resources/sample_onoff.html
@@ -1,0 +1,445 @@
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
+<!-- saved from url=(0028)http://192.168.111.17/?s=1,0 -->
+<html xmlns="http://www.w3.org/1999/xhtml" lang="de" data-lt-installed="true"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
+	<title>STIEBEL ELTRON Reglersteuerung</title>
+	
+	<meta http-equiv="imagetoolbar" content="no">
+	<meta name="content-language" content="de">
+	<meta name="robots" content="noindex,nofollow">
+	<meta name="keywords" content="">
+	<meta name="description" content="">
+  
+  <link rel="shortcut icon" href="http://192.168.111.17/pics/brlogo.ico">
+  <!-- does not exist link rel="stylesheet" href="./css/netzwerk.css" type="text/css" media="all" /-->
+  <link rel="stylesheet" type="text/css" href="./STIEBEL ELTRON Reglersteuerung_files/ste_slider.css">	
+	<!-- Blueprint CSS framework -->
+	<link rel="stylesheet" href="./STIEBEL ELTRON Reglersteuerung_files/screen.css" type="text/css" media="screen, projection">
+	<link rel="stylesheet" href="./STIEBEL ELTRON Reglersteuerung_files/print.css" type="text/css" media="print">	
+	<!--[if lt IE 8]><link rel="stylesheet" href="./css/blueprint/ie.css" type="text/css" media="screen, projection" /><![endif]-->
+	
+	<link rel="stylesheet" href="./STIEBEL ELTRON Reglersteuerung_files/layout.css" type="text/css" media="all">
+  	
+	<link rel="stylesheet" href="./STIEBEL ELTRON Reglersteuerung_files/sifr.css" type="text/css" media="all">
+	<link rel="stylesheet" href="./STIEBEL ELTRON Reglersteuerung_files/thickbox.css" type="text/css" media="all">
+	
+	<link rel="stylesheet" type="text/css" href="./STIEBEL ELTRON Reglersteuerung_files/jquery.jqplot.min.css">
+	
+	  
+  	
+	<script type="text/javascript" src="./STIEBEL ELTRON Reglersteuerung_files/jquery.min.js.Download"></script>
+	<script type="text/javascript" src="./STIEBEL ELTRON Reglersteuerung_files/thickbox.js.Download"></script>
+	<script type="text/javascript" src="./STIEBEL ELTRON Reglersteuerung_files/sifr.js.Download"></script>
+	<script type="text/javascript" src="./STIEBEL ELTRON Reglersteuerung_files/sifr_sw.js.Download"></script>
+	<script type="text/javascript" src="./STIEBEL ELTRON Reglersteuerung_files/scripts.js.Download"></script>
+  <script type="text/javascript" src="./STIEBEL ELTRON Reglersteuerung_files/json2.js.Download"></script>
+  <script type="text/javascript" src="./STIEBEL ELTRON Reglersteuerung_files/jquery-ui-1.8.6.custom.min.js.Download"></script>
+    <script type="text/javascript" src="./STIEBEL ELTRON Reglersteuerung_files/setDNS1.js.Download"></script>
+
+	<!-- [if IE] -->
+	<script language="javascript" type="text/javascript" src="./STIEBEL ELTRON Reglersteuerung_files/excanvas.min.js.Download"></script>
+	<!-- [endif] -->
+
+	<!-- for dateAxisRenderer.js -->
+	<script language="javascript" type="text/javascript" src="./STIEBEL ELTRON Reglersteuerung_files/Date.min.js.Download"></script>
+	<!-- BEGIN: load jqplot -->
+	<script language="javascript" type="text/javascript" src="./STIEBEL ELTRON Reglersteuerung_files/jquery.jqplot.min.js.Download"></script>
+	<script language="javascript" type="text/javascript" src="./STIEBEL ELTRON Reglersteuerung_files/jqplot.dateAxisRenderer.js.Download"></script>
+	<script language="javascript" type="text/javascript" src="./STIEBEL ELTRON Reglersteuerung_files/jqplot.barRenderer.js.Download"></script>
+	<script language="javascript" type="text/javascript" src="./STIEBEL ELTRON Reglersteuerung_files/jqplot.categoryAxisRenderer.js.Download"></script>
+	<!-- END: load jqplot -->
+  
+  <!-- BEGIN: load slider -->
+	<script language="javascript" type="text/javascript" src="./STIEBEL ELTRON Reglersteuerung_files/jquery.ui.core.js.Download"></script>
+	<script language="javascript" type="text/javascript" src="./STIEBEL ELTRON Reglersteuerung_files/jquery.ui.widget.js.Download"></script>
+	<script language="javascript" type="text/javascript" src="./STIEBEL ELTRON Reglersteuerung_files/jquery.ui.mouse.js.Download"></script>
+	<script language="javascript" type="text/javascript" src="./STIEBEL ELTRON Reglersteuerung_files/jquery.ui.slider.js.Download"></script>
+	<!-- END: load slider -->
+  
+    
+  <!-- BEGIN: Uhrzeit -->
+  <script language="javascript" type="text/javascript">
+var timestampunterschied = 1710596767 * 1000 - new Date().getTime();
+</script>
+<script src="./STIEBEL ELTRON Reglersteuerung_files/uhrzeit.js.Download" type="text/javascript" language="javascript"></script>
+  <!-- END:  Uhrzeit -->
+  
+<script src="chrome-extension://nngceckbapebfimnlniiiahkandclblb/content/fido2/page-script.js" id="bw-fido2-page-script"></script></head>
+<body>
+<div id="header">
+	<div class="head curdef">
+		<div id="meta">
+			<div id="login">
+      				Sie sind angemeldet als: <strong>sven</strong>
+        <form method="POST" name="logout" style="display:inline;margin-left:25px;">
+          <input type="submit" alt="Logout" name="logout" value="Logout" class="logoutbutton">
+        </form>
+      			</div>
+		</div>
+    <div class="eingestelle_sprache"><strong><a href="http://192.168.111.17/?s=5,3" style="color:white">DEUTSCH</a></strong></div>
+    <div id="Uhrzeit"><strong>14:46</strong> 16.03.2024</div>
+		<div class="title">
+      <a href="http://192.168.111.17/" onclick="return checkChanges(this);" title="zur Startseite">
+        <img src="./STIEBEL ELTRON Reglersteuerung_files/title.png" alt="Servicewelt">
+		
+      </a>
+            <div style="position: relative; color: #97B765; font-weight: 900; margin: 18px 0px 0px 323px;"></div>
+          </div>
+		<div class="logo"><a href="http://192.168.111.17/" onclick="return checkChanges(this);" title="zur Startseite" class="home_link">&nbsp;</a></div>
+	</div>
+</div>
+
+
+<div class="container" id="page">
+  <div id="main">
+  <div id="SpacerFirefox"></div>
+    <div id="nav">
+      <div class="clear"></div>
+            <div class="nav left defaultfirst">
+        <div class="l left">&nbsp;</div>
+        <div class="c left link1">
+          <a onclick="return checkChanges(this);" href="http://192.168.111.17/?s=0">START</a>
+        </div>
+        <div class="r left">&nbsp;</div>
+        <div class="clear"></div>
+              </div>
+            <div class="nav left default parent">
+        <div class="l left">&nbsp;</div>
+        <div class="c left link2">
+          <a>INFO</a>
+        </div>
+        <div class="r left">&nbsp;</div>
+        <div class="clear"></div>
+                <div class="second">
+          <ul class="second ">
+                        <li>
+			  <a onclick="return checkChanges(this);" href="http://192.168.111.17/?s=1,0">ANLAGE</a>
+    			    
+            </li>
+                      <li>
+			  <a onclick="return checkChanges(this);" href="http://192.168.111.17/?s=1,1">WÄRMEPUMPE</a>
+    			    
+            </li>
+                      <li>
+			  <a onclick="return checkChanges(this);" href="http://192.168.111.17/?s=1,9">Lizenzbestimmungen</a>
+    			    
+            </li>
+                      <li>
+			  <a onclick="return checkChanges(this);" href="http://192.168.111.17/?s=1,10">NUTZUNGS- UND DATENSCHUTZRICHTLINIE</a>
+    			    
+            </li>
+                      <li>
+			  <a onclick="return checkChanges(this);" href="http://192.168.111.17/?s=1,11">HANDBUCH</a>
+    			    
+            </li>
+                    </ul>
+        </div>
+              </div>
+            <div class="nav left default parent">
+        <div class="l left">&nbsp;</div>
+        <div class="c left link3">
+          <a>DIAGNOSE</a>
+        </div>
+        <div class="r left">&nbsp;</div>
+        <div class="clear"></div>
+                <div class="second">
+          <ul class="second ">
+                        <li>
+			  <a onclick="return checkChanges(this);" href="http://192.168.111.17/?s=2,0">STATUS ANLAGE</a>
+    			    
+            </li>
+                      <li>
+			  <a onclick="return checkChanges(this);" href="http://192.168.111.17/?s=2,2">STATUS WÄRMEPUMPE</a>
+    			    
+            </li>
+                      <li>
+			  <a onclick="return checkChanges(this);" href="http://192.168.111.17/?s=2,4">ANALYSE WÄRMEPUMPE</a>
+    			    
+            </li>
+                      <li>
+			  <a onclick="return checkChanges(this);" href="http://192.168.111.17/?s=2,6">INTERNE BERECHNUNG</a>
+    			    
+            </li>
+                      <li>
+			  <a onclick="return checkChanges(this);" href="http://192.168.111.17/?s=2,7">SYSTEM</a>
+    			    
+            </li>
+                      <li>
+			  <a onclick="return checkChanges(this);" href="http://192.168.111.17/?s=2,9">MELDUNGSLISTE</a>
+    			    
+            </li>
+                    </ul>
+        </div>
+              </div>
+            <div class="nav left default parent">
+        <div class="l left">&nbsp;</div>
+        <div class="c left link4">
+          <a>PROGRAMME</a>
+        </div>
+        <div class="r left">&nbsp;</div>
+        <div class="clear"></div>
+                <div class="second">
+          <ul class="second ">
+                        <li>
+			  <a onclick="return checkChanges(this);" href="http://192.168.111.17/?s=3,0">HEIZPROGRAMM</a>
+    			    
+            </li>
+                      <li>
+			  <a onclick="return checkChanges(this);" href="http://192.168.111.17/?s=3,2">WARMWASSERPROGRAMM</a>
+    			    
+            </li>
+                      <li>
+			  <a onclick="return checkChanges(this);" href="http://192.168.111.17/?s=3,4">ZIRKULATIONSPROGRAMM</a>
+    			    
+            </li>
+                      <li>
+			  <a onclick="return checkChanges(this);" href="http://192.168.111.17/?s=3,6">SILENTPROGRAMM 1</a>
+    			    
+            </li>
+                      <li>
+			  <a onclick="return checkChanges(this);" href="http://192.168.111.17/?s=3,7">SILENTPROGRAMM 2</a>
+    			    
+            </li>
+                      <li>
+			  <a onclick="return checkChanges(this);" href="http://192.168.111.17/?s=3,12">PARTYPROGRAMM</a>
+    			    
+            </li>
+                      <li>
+			  <a onclick="return checkChanges(this);" href="http://192.168.111.17/?s=3,13">FERIENPROGRAMM</a>
+    			    
+            </li>
+                      <li>
+			  <a onclick="return checkChanges(this);" href="http://192.168.111.17/?s=3,14">ANTILEGIONELLENPROGRAMM</a>
+    			    
+            </li>
+                    </ul>
+        </div>
+              </div>
+            <div class="nav left default parent">
+        <div class="l left">&nbsp;</div>
+        <div class="c left link5">
+          <a>EINSTELLUNGEN</a>
+        </div>
+        <div class="r left">&nbsp;</div>
+        <div class="clear"></div>
+                <div class="second">
+          <ul class="second ">
+                        <li>
+			  <a onclick="return checkChanges(this);" href="http://192.168.111.17/?s=4,0">ZEIT / DATUM</a>
+    			    
+            </li>
+                      <li>
+			  <a onclick="return checkChanges(this);" href="http://192.168.111.17/?s=4,1">SOMMERZEIT EINSTELLEN</a>
+    			    
+            </li>
+                      <li>
+			  <a onclick="return checkChanges(this);" href="http://192.168.111.17/?s=4,2">HEIZEN</a>
+    			    
+            </li>
+                      <li>
+			  <a onclick="return checkChanges(this);" href="http://192.168.111.17/?s=4,3">WARMWASSER</a>
+    			    
+            </li>
+                      <li>
+			  <a onclick="return checkChanges(this);" href="http://192.168.111.17/?s=4,14">ENERGIEMANAGEMENT</a>
+    			    
+            </li>
+                      <li>
+			  <a onclick="return checkChanges(this);" href="http://192.168.111.17/?s=4,15">ISG</a>
+    			    
+            </li>
+                    </ul>
+        </div>
+              </div>
+            <div class="nav left defaultlast parent">
+        <div class="l left">&nbsp;</div>
+        <div class="c left link6">
+          <a>Profil</a>
+        </div>
+        <div class="r left">&nbsp;</div>
+        <div class="clear"></div>
+                <div class="second">
+          <ul class="second last">
+                        <li>
+			  <a onclick="return checkChanges(this);" href="http://192.168.111.17/?s=5,0">Netzwerk</a>
+    			    
+            </li>
+                      <li>
+			  <a onclick="return checkChanges(this);" href="http://192.168.111.17/?s=5,1">Sicherheit</a>
+    			    
+            </li>
+                      <li>
+			  <a onclick="return checkChanges(this);" href="http://192.168.111.17/?s=5,2">Anlagenbild</a>
+    			    
+            </li>
+                      <li>
+			  <a onclick="return checkChanges(this);" href="http://192.168.111.17/?s=5,3">SPRACHE</a>
+    			    
+            </li>
+                    </ul>
+        </div>
+              </div>
+            <div class="clear"></div>
+    </div>
+        <div id="sub_nav" class="span-24"><div class="left main sifr span-9" style="width: 45%">ANLAGE</div>
+<div class="span-15 last">
+</div>
+<div class="clear"></div></div>
+        <form id="werte" action="http://192.168.111.17/?s=1,0#" onsubmit="saveValues(this);return false;"><div id="content">
+    <div class="span-11 append-1" style="float:left"><table class="info"><tbody><tr><th colspan="2" class="round-top">HEIZUNG</th></tr>  <tr class="even">
+    <td class="key">AUSSENTEMPERATUR</td>
+    <td class="value">10,6°C</td>
+  </tr>
+  <tr class="odd">
+    <td class="key">ISTTEMPERATUR HK 1</td>
+    <td class="value">33,2°C</td>
+  </tr>
+  <tr class="even">
+    <td class="key">SOLLTEMPERATUR HK 1</td>
+    <td class="value">31,8°C</td>
+  </tr>
+  <tr class="odd">
+    <td class="key">VORLAUFISTTEMPERATUR WP</td>
+    <td class="value">42,8°C</td>
+  </tr>
+  <tr class="even">
+    <td class="key">VORLAUFISTTEMPERATUR NHZ</td>
+    <td class="value">42,6°C</td>
+  </tr>
+  <tr class="odd">
+    <td class="key">RÜCKLAUFISTTEMPERATUR WP</td>
+    <td class="value">36,6°C</td>
+  </tr>
+  <tr class="even">
+    <td class="key">FESTWERTBETRIEB</td>
+    <td class="value">Aus</td>
+  </tr>
+  <tr class="odd">
+    <td class="key">PUFFERISTTEMPERATUR</td>
+    <td class="value">33,2°C</td>
+  </tr>
+  <tr class="even">
+    <td class="key">PUFFERSOLLTEMPERATUR</td>
+    <td class="value">31,8°C</td>
+  </tr>
+  <tr class="odd">
+    <td class="key">HEIZUNGSDRUCK</td>
+    <td class="value">1,21bar</td>
+  </tr>
+  <tr class="even">
+    <td class="key">VOLUMENSTROM</td>
+    <td class="value">8,98l/min</td>
+  </tr>
+  <tr class="odd">
+    <td class="key round-leftbottom">FROSTSCHUTZ</td>
+    <td class="value round-rightbottom">4,0°C</td>
+  </tr>
+</tbody></table></div><div class="span-11 prepend-1" style="float:right"><table class="info"><tbody><tr><th colspan="2" class="round-top">WARMWASSER</th></tr>  <tr class="even">
+    <td class="key">ISTTEMPERATUR</td>
+    <td class="value">49,3°C</td>
+  </tr>
+  <tr class="odd">
+    <td class="key">SOLLTEMPERATUR</td>
+    <td class="value">48,0°C</td>
+  </tr>
+  <tr class="even">
+    <td class="key round-leftbottom">VOLUMENSTROM</td>
+    <td class="value round-rightbottom">8,98l/min</td>
+  </tr>
+</tbody></table></div><div class="span-11 prepend-1" style="float:right"><table class="info"><tbody><tr><th colspan="2" class="round-top">ELEKTRISCHE NACHERWÄRMUNG</th></tr>  <tr class="even">
+    <td class="key">BIVALENZTEMPERATUR HZG</td>
+    <td class="value">-25,0°C</td>
+  </tr>
+  <tr class="odd">
+    <td class="key">UNTERE EINSATZGRENZE HZG</td>
+    <td class="value">-25,0°C</td>
+  </tr>
+  <tr class="even">
+    <td class="key">BIVALENZTEMPERATUR WW</td>
+    <td class="value">-15,0°C</td>
+  </tr>
+  <tr class="odd">
+    <td class="key round-leftbottom">UNTERE EINSATZGRENZE WW</td>
+    <td class="value round-rightbottom">Aus</td>
+  </tr>
+</tbody></table></div><div class="span-11 prepend-1" style="float:right"><table class="info"><tbody><tr><th colspan="2" class="round-top">ENERGIEMANAGEMENT</th></tr>  <tr class="even">
+    <td class="key">SG READY 1</td>
+    <td class="value">0</td>
+  </tr>
+  <tr class="odd">
+    <td class="key">SG READY 2</td>
+    <td class="value">0</td>
+  </tr>
+  <tr class="even">
+    <td class="key">Betriebszustand 2</td>
+    <td class="value"><img height="15" src="./STIEBEL ELTRON Reglersteuerung_files/ste-symbol_an-97b765.png"></td>
+  </tr>
+</tbody></table></div><div class="span-24">&nbsp;</div>
+    </div>  </form></div>
+</div>
+
+
+<div id="footer">
+	<!--[if IE]>
+	<style type="text/css">
+	#footer
+{
+		
+		height:260px;
+}
+	#versionsNummer
+	{
+		position:relative;
+		top:-33px;
+	}
+	</style>
+
+     	<![endif]-->
+	<div class="span-24 last footer_middle curdef">
+	
+		<div class="span-8 prepend-top" style="height:145px">
+						<div class="span-3">
+			
+				<img src="./STIEBEL ELTRON Reglersteuerung_files/ste_footer_website.png">
+			</div>
+			<div class="span-5 last">
+				<p>
+				
+          					Weitere Informationen zu unserem Produktprogramm und aktuelle Serviceangebote finden Sie unter					<a href="http://www.stiebel-eltron.de/" target="_blank" class="extern">www.stiebel-eltron.de</a>
+				</p>
+			</div>
+					</div>
+	  
+    		<div class="bordered span-1"></div>
+		<div class="span-6 prepend-top">
+			<div class="span-2">
+			  <img src="./STIEBEL ELTRON Reglersteuerung_files/footer_tel.png">
+			</div>
+			<div class="span-4 last">
+				<p>
+					Unsere Service-Hotline erreichen Sie unter<br><span class="big">05531 702-111</span>
+				</p>
+			</div>
+	  </div>
+    		<div class="bordered span-1"></div>
+		<div class="span-7 prepend-top last">
+			<p>
+				Das Neueste rund um die Servicewelt				<br><a href="https://www.stiebel-eltron.de/de/home/service/smart-home/servicewelt.html" target="_blank" class="extern">servicewelt</a>
+			</p>
+	  </div>
+    
+	</div>
+  <div class="links span-23 prepend-top append-1 last footer_middle">
+    <div class="span-17">
+        		<a class="intern" href="http://192.168.111.17/?forcepage=dsgvo" onclick="return checkChanges(this);">Datenschutz</a>
+				
+		<a href="http://www.stiebel-eltron.de/de/info/impressum.html">Impressum</a>
+      &nbsp;
+      <p style="margin-top:8px" class="right" id="versionsNummer">v12.2.3</p>
+    </div>
+  	<div class="copyright span-6 last curdef">
+  		© STIEBEL ELTRON 2024  	</div>
+  </div>
+</div>
+
+
+</body></html>

